packages/jellyfish-transaction/src/script/defi/dftx_governance.ts

```diff
@@ -0,0 +1,91 @@
+import BigNumber from 'bignumber.js'
+import { SmartBuffer } from 'smart-buffer'
+import { readBigNumberUInt64, writeBigNumberUInt64 } from '../../buffer/buffer_bignumber'
+import { BufferComposer, ComposableBuffer } from '../../buffer/buffer_composer'
+
+// Disabling no-return-assign makes the code cleaner with the setter and getter */
+/* eslint-disable no-return-assign */
+
+export interface LiqPoolSplit {
+  tokenId: number // -------------------| 4 bytes unsigned
+  value: BigNumber // ------------------| 8 bytes unsigned
+}
+
+export interface GovernanceVar {
+  key: string // -----------------------| VarUInt{1-9 bytes}
+  value: BigNumber | LiqPoolSplit[] // -| VarUInt{8 OR 1 + n * 12 bytes}, case LiqPoolSplit: first byte = array len
+}
+
+export interface SetGovernance {
+  governanceVars: GovernanceVar[]
+}
+
+/**
+ * Composable TokenBalance, C stands for Composable.
+ * Immutable by design, bi-directional fromBuffer, toBuffer deep composer.
+ */
+export class CSetGovernance extends ComposableBuffer<SetGovernance> {
+  static OP_CODE = 0x47 // 'G'
+  static OP_NAME = 'OP_DEFI_TX_SET_GOVERNANCE'
+
+  composers (gvs: SetGovernance): BufferComposer[] {
+    return [
+      {
+        fromBuffer: (buffer: SmartBuffer): void => {
+          gvs.governanceVars = []
+          while (buffer.remaining() > 0) {
+            const keyLen = buffer.readUInt8()
```

Comment:
Main issue causing it to require manual rollover, missing the length of array `GovernanceVar` to be set.